src/components/Waveform/Waveform.tsx

```diff
@@ -136,11 +137,13 @@ export const Waveform = forwardRef<IWaveformRef, IWaveform>((props, ref) => {
   const getAudioWaveFormForPath = async (noOfSample: number) => {
     if (!isNil(path) && !isEmpty(path)) {
       try {
+        (onWaveformLoadStateChange as Function)?.(true);
         const result = await extractWaveformData({
           path: path,
           playerKey: `PlayerFor${path}`,
           noOfSamples: noOfSample,
         });
+        (onWaveformLoadStateChange as Function)?.(false);
```

Comment:
We should add this line to catch block as well. If there is an error while extracting waveform data, then this line will not be executed, so we need to send a callback from the catch block that loading is done.
In the catch block, we have an error callback; from there, they can identify the error.

Reply:
Resolved, Please Check